docs/api/serviceregistry/service-registry-management-generic-mqtt.md

```diff
@@ -0,0 +1,1728 @@
+
+```
+{
+  "status": 400,
+  "traceId": "<trace-id>",
+  "receiver": "sysop",
+  "payload": "Device with names already exists: alarm1, alarm2"
+}
+```
+
+### device-update
+
+The service operation **request** requires an [MQTTRequestTemplate](../data-models/mqtt-request-template.md) JSON encoded message in which the authentication is a proper [identity info](../../api/authentication_policy.md/#mqtt) and the payload is a [DeviceListRequest](../data-models/device-list-request.md).
+
+```
+Topic: arrowhead/serviceregistry/management/device-update
+
+{
+  "traceId": "<trace-id>",
+  "authentication": "<identity-info>",
+  "responseTopic": "<response-topic>",
+  "qosRequirement": <0|1|2>,
+  "payload": {
+    "devices": [
+      {
+        "name": "alarm1",
+        "metadata": {
+          "volume": { "value": 100, "unit": "dB"}
+        },
+        "addresses": [
+          "4a:f7:9c:12:8e:b5"
+        ]
+      },
+      {
+        "name": "alarm2",
+        "metadata": {
+          "volume": { "value": 110, "unit": "dB"}
+        },
+        "addresses": [
+          "4a:f7:9c:12:8e:bb"
+        ]
+      }
+    ]
+  }
+}
+```
+
+The service operation **responds** with an [MQTTResponseTemplate](../data-models/mqtt-response-template.md) JSON encoded message in which the status code is `200` if called successfully. The response template payload is a [DeviceListResponse](../data-models/device-list-response.md).
+
+```
+{
+  "status": 200,
+  "traceId": "<trace-id>",
+  "receiver": "sysop",
+  "payload": {
+    "entries": [
+      {
+        "name": "alarm1",
+        "metadata": {
+          "volume": {
+            "value": 100,
+            "unit": "dB"
+          }
+        },
+        "addresses": [
+          {
+            "type": "MAC",
+            "address": "4a:f7:9c:12:8e:b5"
+          }
+        ],
+        "createdAt": "2025-05-04T18:51:46Z",
+        "updatedAt": "2025-05-04T18:51:46Z"
+      },
+      {
+        "name": "alarm2",
+        "metadata": {
+          "volume": {
+            "value": 110,
+            "unit": "dB"
+          }
+        },
+        "addresses": [
+          {
+            "type": "MAC",
+            "address": "4a:f7:9c:12:8e:bb"
+          }
+        ],
+        "createdAt": "2025-05-04T18:51:46Z",
+        "updatedAt": "2025-05-04T18:51:46Z"
+      }
+    ],
+    "count": 2
+  }
+}
+```
+
+The **error codes** are `400` if the request is malformed, `401` if the requester authentication was unsuccessful, `403` if the authenticated requester has no permission and `500` if an unexpected error happens. In these cases the response template payload is an [ErrorResponse](../data-models/error-response.md) JSON.
```

Comment:
There is also a 423 when you try to delete a device with systems already assigned to it (This could be an error in the HTTP IDD as well)